src/apm/java.go

```diff
@@ -55,28 +87,35 @@ func InjectJavaagent(javaSpec v1alpha1.Java, pod corev1.Pod, index int) (corev1.
 		container.Env[idx].Value = container.Env[idx].Value + javaJVMArgument
 	}
 
-	container.VolumeMounts = append(container.VolumeMounts, corev1.VolumeMount{
-		Name:      volumeName,
-		MountPath: "/newrelic-instrumentation",
-	})
+	if isContainerVolumeMissing(container, volumeName) {
+		container.VolumeMounts = append(container.VolumeMounts, corev1.VolumeMount{
+			Name:      volumeName,
+			MountPath: "/newrelic-instrumentation",
+		})
+	}
 
 	// We just inject Volumes and init containers for the first processed container.
-	if isInitContainerMissing(pod) {
-		pod.Spec.Volumes = append(pod.Spec.Volumes, corev1.Volume{
-			Name: volumeName,
-			VolumeSource: corev1.VolumeSource{
-				EmptyDir: &corev1.EmptyDirVolumeSource{},
-			}})
+	if isInitContainerMissing(pod, javaInitContainerName) {
+		if isPodVolumeMissing(pod, volumeName) {
+			pod.Spec.Volumes = append(pod.Spec.Volumes, corev1.Volume{
+				Name: volumeName,
+				VolumeSource: corev1.VolumeSource{
+					EmptyDir: &corev1.EmptyDirVolumeSource{},
+				}})
+		}
 
 		pod.Spec.InitContainers = append(pod.Spec.InitContainers, corev1.Container{
-			Name:    initContainerName,
-			Image:   javaSpec.Image,
+			Name:    javaInitContainerName,
+			Image:   inst.Spec.Agent.Image,
 			Command: []string{"cp", "/newrelic-agent.jar", "/newrelic-instrumentation/newrelic-agent.jar"},
 			VolumeMounts: []corev1.VolumeMount{{
 				Name:      volumeName,
 				MountPath: "/newrelic-instrumentation",
 			}},
 		})
 	}
+
+	pod = i.injectNewrelicConfig(ctx, inst.Spec.Resource, ns, pod, index, inst.Spec.LicenseKeySecret)
+
```

Comment:
nit: can you pls change the below to `return pod, nil`

Reply:
fixed